### ISSUE TITLE
For file() table function allow partial structure definition with placeholders.

### ISSUE THREAD

**Issue by yakov-olkhovskiy** [ClickHouse/ClickHouse#48074]
Structure definition argument in file() function allows to predefine columns with their types instead structure inference. Proposed feature should allow use columns names (placeholders) without type in structure definition - for such columns type will be inferred from actual data.
Example:
```
SELECT * FROM file('data.csv', auto, 'a UInt32, b, c, d String')
```

**Comment by kssenii** [comment]
It is already implemented https://github.com/ClickHouse/ClickHouse/pull/40068, though looks a bit different

**Comment by yakov-olkhovskiy** [comment]
Implemented in #40068 as settings `schema_inference_hints` and `column_names_for_schema_inference`
